src/common/keyboard_layout.h

```diff
@@ -0,0 +1,20 @@
+#pragma once

```

Comment:
I don't know much about this, but can you share some reference about this procedure? Like why do we need a separate Impl class which is referenced by a pointer?

Reply:
The reason this is necessary is because projects compiled with the `\clr` flag don't support `std::mutex`. Therefore, we need to be careful not to include the `<mutex>` header in the header for the exported classes, otherwise the compiler will complain. So what we can do is to keep two header files, one public and one internal, and with a public and a private class. The public class then keeps an opaque pointer to the internal class, by forward declaring the class, and in the cpp file we implement both the public and the internal class.

There is more info here https://en.wikipedia.org/wiki/Opaque_pointer#C++